mongo_orchestration/sharded_clusters.py

```diff
@@ -56,8 +56,24 @@ def __init__(self, params):
         if self.sslParams:
             self.kwargs.update(DEFAULT_SSL_OPTIONS)
 
+        # Determine what to do with config servers via mongos version.
+        mongos = Server(name='mongos', procParams={})
+        mongos_version = mongos.version
+        mongos.cleanup()
         configsvr_configs = params.get('configsvrs', [{}])
-        self.__init_configsvr(configsvr_configs)
+        self.uses_rs_configdb = (mongos_version >= (3, 1, 2) and
+                                 len(configsvr_configs) == 1)
+        self.configdb_singleton = (
+            ReplicaSets() if self.uses_rs_configdb else Servers())
+        if self.uses_rs_configdb:
+            self.__init_configrs(configsvr_configs[0])
+        elif mongos_version >= (3, 1, 2) and len(configsvr_configs) != 3:
```

Comment:
Also need to return a useful error message when the server version is >= 3.3.2 and 3 old style config servers were specified.